mxcubeweb/core/adapter/beamline_action_adapter.py

```diff
@@ -1,21 +1,39 @@
+from typing import ClassVar
+
 import logging
 from enum import Enum
 
-from mxcubeweb.core.adapter.adapter_base import ActuatorAdapterBase
+from mxcubecore.HardwareObjects.BeamlineActions import BeamlineActions
+from mxcubecore.BaseHardwareObjects import HardwareObject
+
+from mxcubeweb.core.models.configmodels import AdapterResourceHandlerConfigModel
+
+from mxcubeweb.core.adapter.adapter_base import AdapterBase
 from mxcubeweb.core.models.adaptermodels import (
     HOActuatorValueChangeModel,
     NStateModel,
-    StrValueModel,
 )
 
+resource_handler_config = AdapterResourceHandlerConfigModel(
+    commands=["stop"], attributes=["data"]
+)
+
+
+class BeamlineActionAdapter(AdapterBase):
+    SUPPORTED_TYPES: ClassVar[list[object]] = [BeamlineActions]
 
-class BeamlineActionAdapter(ActuatorAdapterBase):
-    def __init__(self, ho, *args, **kwargs):
+    def __init__(
+        self,
+        ho: HardwareObject,
+        role: str,
+        app,
```

Comment:
I can't import MXCUBEApp due to circular import, should we use Type["MXCUBEApp"] in those cases, or do you think we should deal with such cases ?